pkg/common/common.go

```diff
@@ -14,10 +14,20 @@
 
 package common
 
+import (
+	operator "github.com/tigera/operator/api/v1"
+)
+
 const (
 	CalicoNamespace     = "calico-system"
 	TyphaDeploymentName = "calico-typha"
 	NodeDaemonSetName   = "calico-node"
 
 	TigeraPrometheusNamespace = "tigera-prometheus"
 )
+
+type InstallationInternal struct {
+	Spec                *operator.InstallationSpec
```

Comment:
One last idea - would it make sense to use struct inheritence so we don't have to pass `.Spec` everywhere?
```suggestion
	*operator.InstallationSpec
```

Reply:
I could make this change but for all the render calls that take an InstallationSpec we would need to pass the embedded struct so embedding it would not make a difference.
For example
```
x := common.InstallationInternal

// Render call taking operator.InstallationSpec
render.Example(x.InstallationSpec)
```

Reply:
Is the idea here to wrap an installation spec with additional fields?

Aren't `TigeraCustom` and `NodeAppArmorProfile` derivable from the "real" Installation API? I'm not sure I understand the purpose of this structure.

Reply:
We stopped passing the full Installation resource around because really renders should (ideally) only need the Spec. Also with the overlay functionality we only needed to merge the InstallationSpec so it made since to stop passing the full Installation around.

Reply:
Right, but this PR is re-introducing the need to pass more than just the spec to render.

It feels wrong to then say "Render only needs spec, but there's also this other field that we want to pass, so we'll create a new resource that just tacks on the new field to the Installation spec".

I'd rather remove this structure altogether, and then either:
- Pass the whole Installation again, because that is what is needed.
- Extend the render functions to take an additional argument wherever needed, passing in the relevant fields explicitly.